pkg/manager/internal.go

```diff
@@ -348,6 +365,13 @@ func (cm *controllerManager) serveMetrics(stop <-chan struct{}) {
 	// TODO(JoelSpeed): Use existing Kubernetes machinery for serving metrics
 	mux := http.NewServeMux()
 	mux.Handle(metricsPath, handler)
+
+	cm.mu.Lock()
```

Comment:
Do we need the lock at this point? I'd assume it's safe to read when `serveMetrics` is called. If we do, I'd use 
```
cm.mu.Lock()
defer cm.mu.Unlock()
```
pattern for consistency with the other methods

Reply:
Normally we don't need the lock, but I think it is theoretically possible to trigger race condition if `Start`-ing and `AddMetricsExtraHandler`-ing at the same time. So probably it's better just to have the mutex.

Regarding

```
cm.mu.Lock()
defer cm.mu.Unlock()
```

It is not suitable for this case, because `serveMetrics` blocks on receiving from `stop` channel thus the method does not return until the manager is stopped. This is the reason `serveHealthProbes` also does not follow the pattern.

Reply:
Think you can wrap the locking/unlocking and the loop inside a `func` and use `defer` there. That ensures that even if something bad happens during the loop, lock is released. 

```go
func() {
    c.mu.Lock()
    defer c.mu.Unlock()
    for path, extraHandler := range cm.metricsExtraHandlers {
        mux.Handle(path, extraHandler)
    }
}()
```

Reply:
Nice tip, thanks! Changed according to your snippet.

Reply:
@hypnoglow Could you also a TODO comment to make sure we do this in `serveHealthProbes` function too? Thanks.

Reply:
Do you suggest adding this comment in `serveHealthProbes` or in `serveMetrics`?

Reply:
Wherever applicable, please.

Reply:
Added comment to `serveHealthProbes`, thanks.